src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -370,7 +375,20 @@ public void HandleIncomingRequest(Message message, ActivationData targetActivati
         {
             lock (targetActivation)
             {
-                if (targetActivation.State == ActivationState.Invalid)
+                if (targetActivation.Grain.IsGrain)
+                {
+                    var request = ((InvokeMethodRequest)message.GetDeserializedBody(this.serializationManager));
```

Comment:
Because interface version is part of `InvokeMethodRequest`, we have to deserialize message body before we can decide what to do with the message, potentially to forward it to another silos. In such a case this serialization is wasteful.

Could we instead put interface version into message headers to avoid deserialization here? That would also keep messages backward and forward compatible.

Going one step further (out of scope of this PR), should we consider moving `InterfaceId` and `MethodId` from `InvokeMethodRequest` to headers, and leave only arguments in the body?